fractions/javaee/undertow/src/main/java/org/wildfly/swarm/undertow/runtime/ContextPathArchivePreparer.java

```diff
@@ -15,29 +15,66 @@
  */
 package org.wildfly.swarm.undertow.runtime;
 
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.logging.Logger;
+
 import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
 
 import org.jboss.shrinkwrap.api.Archive;
 import org.wildfly.swarm.spi.api.ArchivePreparer;
 import org.wildfly.swarm.spi.api.Defaultable;
 import org.wildfly.swarm.spi.api.annotations.Configurable;
+import org.wildfly.swarm.spi.api.config.ConfigKey;
+import org.wildfly.swarm.spi.api.config.ConfigView;
+import org.wildfly.swarm.spi.api.config.SimpleKey;
 import org.wildfly.swarm.undertow.WARArchive;
+import org.wildfly.swarm.undertow.internal.UndertowExternalMountsAsset;
 
 /**
  * @author Ken Finnigan
  */
 @ApplicationScoped
 public class ContextPathArchivePreparer implements ArchivePreparer {
 
+    @SuppressWarnings("unused")
+    private static final Logger log = Logger.getLogger(ContextPathArchivePreparer.class.getName());
+
     @Configurable("swarm.context.path")
     Defaultable<String> contextPath = Defaultable.string("/");
 
+    @Inject
+    ConfigView configView;
+
     @Override
     public void prepareArchive(Archive<?> archive) {
         WARArchive warArchive = archive.as(WARArchive.class);
 
         if (warArchive.getContextRoot() == null) {
             warArchive.setContextRoot(contextPath.get());
         }
+
+        UndertowExternalMountsAsset ut = null;
+        ConfigKey parentKey = ConfigKey.parse("swarm.context.mount");
+        if (configView != null && configView.hasKeyOrSubkeys(parentKey)) {
+            for (SimpleKey mount : configView.simpleSubkeys(parentKey)) {
```

Comment:
In a recent commit, we added the ability to inject a @Configurable such as

    @Configurable("swarm.context.mount")
    Map<String,?> mounts

Which may allow, if it's not buggy, for you to then iterate, where the keys are the immediate subkey, and the value is the subtree from the YAML or whatnot, in your case a simple `String` I think.

Might be worth trying, as it might simplify your usage here.

A more complex example (involving a List of Maps from the YAML) is here:

https://github.com/wildfly-swarm/wildfly-swarm/blob/c60fb01a407b41ecab86e510dbba3d4ff25c1a65/core/container/src/test/java/org/wildfly/swarm/container/config/ConfigViewYamlTest.java